internal/sysfs/select_windows_test.go

```diff
@@ -2,40 +2,66 @@ package sysfs
 
 import (
 	"context"
+	"net"
 	"os"
 	"syscall"
 	"testing"
 	"time"
 
+	"github.com/tetratelabs/wazero/experimental/sys"
+	"github.com/tetratelabs/wazero/internal/platform"
 	"github.com/tetratelabs/wazero/internal/testing/require"
 )
 
 func TestSelect_Windows(t *testing.T) {
 	type result struct {
-		hasData bool
-		err     error
+		n     int
+		fdSet platform.FdSet
+		err   sys.Errno
 	}
 
 	testCtx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
+	handleAsFdSet := func(readHandle syscall.Handle) *platform.FdSet {
+		var fdSet platform.FdSet
+		fdSet.Set(int(readHandle))
+		return &fdSet
+	}
+
 	pollToChannel := func(readHandle syscall.Handle, duration *time.Duration, ch chan result) {
 		r := result{}
-		r.hasData, r.err = pollNamedPipe(testCtx, readHandle, duration)
+		fdSet := handleAsFdSet(readHandle)
+		r.n, r.err = selectAllHandles(testCtx, fdSet, nil, nil, duration)
+		r.fdSet = *fdSet
 		ch <- r
 		close(ch)
 	}
 
+	t.Run("syscall_select returns sys.ENOSYS when n == 0 and duration is nil", func(t *testing.T) {
+		n, errno := syscall_select(0, nil, nil, nil, nil)
+		require.Equal(t, -1, n)
+		require.EqualErrno(t, sys.ENOSYS, errno)
+	})
+
+	t.Run("syscall_select propagates error when peekAllPipes returns an error", func(t *testing.T) {
+		fdSet := platform.FdSet{}
+		fdSet.Pipes().Set(-1)
+		n, errno := syscall_select(0, &fdSet, nil, nil, nil)
+		require.Equal(t, -1, n)
+		require.EqualErrno(t, sys.ENOSYS, errno)
+	})
+
 	t.Run("peekNamedPipe should report the correct state of incoming data in the pipe", func(t *testing.T) {
 		r, w, err := os.Pipe()
 		require.NoError(t, err)
 		rh := syscall.Handle(r.Fd())
 		wh := syscall.Handle(w.Fd())
 
-		// Ensure the pipe has data.
+		// Ensure the pipe has no data.
 		n, err := peekNamedPipe(rh)
```

Comment:
nit if errno return as the name errno and then use require.EqualErrno as it has better formatting on fail